templates/lxc-local.in

```diff
@@ -144,130 +161,222 @@ relevant_file() {
+    cat "${LXC_PATH}/config-extra"
   fi
-} >> "${LXC_PATH}/config"
+}
 
-fstab="$(relevant_file fstab)"
-if [ -e "${fstab}" ]; then
-  echo "lxc.mount.fstab = ${LXC_PATH}/fstab" >> "${LXC_PATH}/config"
-fi
-echo "lxc.uts.name = ${LXC_NAME}" >> "${LXC_PATH}/config"
+add_container_config() {
+  # Adds the contents of the supplied metadata config to the container config
+  {
+    echo ""
+    echo "# Distribution configuration"
+    cat "$configfile"
+  } >> "${LXC_PATH}/config"
+}
 
-## Re-add the previously removed network config
-if [ -e "${LXC_PATH}/config-network" ]; then
+add_extra_config() {
+  # Add the container-specific config
   {
     echo ""
-    echo "# Network configuration"
-    cat "${LXC_PATH}/config-network"
-    rm "${LXC_PATH}/config-network"
+    echo "# Container specific configuration"
+    if [ -e "${LXC_PATH}/config-extra" ]; then
+      cat "${LXC_PATH}/config-extra"
+      rm "${LXC_PATH}/config-extra"
+    fi
   } >> "${LXC_PATH}/config"
-fi
+}
 
-TEMPLATE_FILES="${LXC_PATH}/config"
+add_network_config() {
+  # Re-add the previously removed network config
+  if [ -e "${LXC_PATH}/config-network" ]; then
+    {
+      echo ""
+      echo "# Network configuration"
+      cat "${LXC_PATH}/config-network"
+      rm "${LXC_PATH}/config-network"
+    } >> "${LXC_PATH}/config"
+  fi
+}
 
-# Setup the fstab
-if [ -e "${fstab}" ]; then
-  cp "${fstab}" "${LXC_PATH}/fstab"
-  TEMPLATE_FILES="${TEMPLATE_FILES};${LXC_PATH}/fstab"
-fi
+process_config() {
+  # Process the supplied config file if it exists.
+  configfile="$(relevant_file config)"
+  if [ ! -e "${configfile}" ]; then
+    echo "ERROR: metadata tarball is missing the configuration file" 1>&2
+    return
+  fi
+  extract_config
+  add_container_config
+  add_extra_config
+  add_network_config
+}
 
-# Look for extra templates
-if [ -e "$(relevant_file templates)" ]; then
-  while read -r line; do
-    fullpath="${LXC_ROOTFS}/${line}"
-    [ ! -e "${fullpath}" ] && continue
-    TEMPLATE_FILES="${TEMPLATE_FILES};${fullpath}"
-  done < "$(relevant_file templates)"
-fi
+process_fstab() {
+  # Process the supplied fstab file if it exists.
+  # Add the fstab file to template files so substitions can be made.
+  fstab="$(relevant_file fstab)"
+  if [ -e "${fstab}" ]; then
+    echo "lxc.mount.fstab = ${LXC_PATH}/fstab" >> "${LXC_PATH}/config"
+    cp "${fstab}" "${LXC_PATH}/fstab"
+    TEMPLATE_FILES="${TEMPLATE_FILES};${LXC_PATH}/fstab"
+  fi
+}
 
+process_templates() {
+  # Look for extra templates
+  template="$(relevant_file template)"
+  if [ -e "${template}" ]; then
+    while read -r line; do
+      fullpath="${LXC_ROOTFS}/${line}"
+      [ ! -e "${fullpath}" ] && continue
+      TEMPLATE_FILES="${TEMPLATE_FILES};${fullpath}"
+    done < "${template}"
+  fi
+}
+
+unpack_metadata() {
+  # Unpack file that contains the container metadata
+  # If the file does not exist, just warn and continue.
+  if [ -n "${LXC_METADATA}" ]; then
```

Comment:
Both of those if statements should still get the indent fix.
Making the function immediately return if the medata file isn't passed or if it doesn't exist.